test/controllers/samples_controller_test.rb

```diff
@@ -233,6 +235,13 @@ class SamplesControllerTest < ActionDispatch::IntegrationTest
     end
   end
 
+  test 'joe cannot fetch pipeline stage results for another user\'s private samples' do
```

Comment:
Can you add a test for the feature flag, i.e. does not see if feature flag not enabled?